src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/Assumed.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Razor;
+
+internal static class Assumed
```

Comment:
Cyrus added information to the roslyn version of this to improve debuggability. We should probably just use the same technique he used: https://github.com/dotnet/roslyn/pull/64402

Reply:
Here's a screenshot of the important part of the diff in that PR, as it's very big:
![image](https://github.com/dotnet/razor/assets/2371880/45941af1-178c-4699-9370-4c907642bd7c)

Reply:
Great idea! I'll steal it.